haskell/test/SMT/Model/BaseDAO/Plist.hs

```diff
@@ -0,0 +1,130 @@
+-- SPDX-FileCopyrightText: 2021 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+-- | Common types between different DAO implementations.
+module SMT.Model.BaseDAO.Plist
+  ( plistMem
+  , plistInsert
+  , plistDelete
+  , plistUpdateHead
+  ) where
+
+import Universum
+
+import qualified Data.Map as Map
+
+import Morley.Michelson.Typed.Haskell.Value (BigMap(..))
+
+import Ligo.BaseDAO.Types
+
+plistMem :: ProposalKey -> Maybe ProposalDoublyLinkedList -> Bool
+plistMem key plistMb =
+  case plistMb of
+    Just ProposalDoublyLinkedList{..} ->
+      (plFirst == key) ||
+      (case Map.lookup (key, False) (plMap & bmMap) of
+          Nothing -> False
+          Just _ -> True
+      )
+    Nothing -> False
+
+
+
+plistInsert :: ProposalKey -> Maybe ProposalDoublyLinkedList -> Maybe ProposalDoublyLinkedList
+plistInsert key plistMb =
+  case plistMb of
+    Nothing ->
+      Just $ ProposalDoublyLinkedList { plFirst = key, plLast = key, plMap = mempty }
+    Just plist ->
+      Just $ plist
+        { plLast = key
+        , plMap = (plist & plMap & bmMap)
+            & Map.insert (key, False) (plist & plLast)
+            & Map.insert ((plist & plLast), True) key
+            & BigMap Nothing
+        }
+
+
+plistDelete :: ProposalKey -> Maybe ProposalDoublyLinkedList -> Maybe ProposalDoublyLinkedList
+plistDelete key plistMb =
+  case plistMb of
+    Nothing ->
+      error "BAD_STATE: no proposal"
+    Just plist ->
+      if (plist & plFirst) == key then
```

Comment:
Is there any advantage in using `&` instead of simply writing `plFirst plist`?

Just want to share that I find it a bit harder than I would like to admit, to read this code even when I understand how `&` works.

Reply:
I choose to do since it resembles `plist.first` in ligo, which makes it easier to compare the ligo and haskell implementation.